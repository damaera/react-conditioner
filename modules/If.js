import React, { PropTypes } from 'react';

const If = (props) => {
  if (props.condition) {
    return (
      <span>
        { props.children }
      </span>
    )
  } else {
    return null
  }
}

If.propTypes = {
  condition: PropTypes.any,
  children: PropTypes.element,
}

export default If;