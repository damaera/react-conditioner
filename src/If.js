import React from 'react';

import PropTypes from 'prop-types';

const If = (props) => {
  if (props.condition) {
    return (
      <span>
        { props.children }
      </span>
    )
  }
  return null
}

If.propTypes = {
  condition: PropTypes.any,
  children: PropTypes.any,
}

export default If;