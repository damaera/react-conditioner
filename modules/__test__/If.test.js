import React from 'react';
import renderer from 'react-test-renderer';

import If from '../If'

it('render string when true', () => {
  const tree = renderer.create(
    <If condition={true}>
      Test
    </If>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('render element when true', () => {
  const tree = renderer.create(
    <If condition={true}>
      <span>sdfsdf</span>
    </If>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('not render when false', () => {
  const tree = renderer.create(
    <If condition={false}>
      <span>sdfsdf</span>
    </If>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
