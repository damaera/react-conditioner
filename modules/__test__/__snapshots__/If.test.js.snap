// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not render when false 1`] = `null`;

exports[`render element when true 1`] = `
<span>
  <span>
    sdfsdf
  </span>
</span>
`;

exports[`render string when true 1`] = `
<span>
  Test
</span>
`;
